### DISCUSSION TITLE
Field with readonly and default field not saving

### DISCUSSION THREAD

**Discussion by internetbug256** [laravel/nova-issues#3667] [Q&A]
Hi there. 
I had to open a new discussion with the same title as https://github.com/laravel/nova-issues/issues/3630 because that one was immediately closed.
I strongly disagree with the conclusion. Why a read-only field won't be saved??
Is that REALLY the expected behavior?
In my case, I need to show in a create form that the field will be saved with that specific value, and it cannot be edited.

Thanks

**Comment by GerwinV** [comment]
I got the same thing. I want the field to be read-only with a default value for some users, but admins are allowed to change the value.
I want to show the value to make sure users see what the value will be, so they dont only see it on the index of show view.

**Comment by evan10s** [comment]
I ran into this just now and was able to get around it with a hacky way (which seems like an [old way](https://stackoverflow.com/a/52614335) of making fields read-only) of just adding the read-only attribute to the select (in my case) field:

```php
Select::make('field name')
->withMeta(['extraAttributes' => ['readonly' => true]])
```

instead of

```php
Select::make('field name')
->readonly()
```

**Comment by RealityGamin** [comment]
You can always override Nova behavior from Laravel Model Observers using events, if you require more information, I can help you with more code. https://laravel.com/docs/9.x/eloquent#observers 

But in general, this has worked for me without any issues to give you an idea on different approach perhaps, try to replicate and let us know: (Nova 4), but in general, observers can accomplish what you are looking for.

```
Select::make('Type')->options([
                '0' => 'AA',
                '1' => 'BB',
                '2' => 'CC',
                '3' => 'DD',
            ])->displayUsingLabels()->rules('required')->hideFromIndex()->searchable(),

Select::make('QR Code Registration', 'self_registration')->options([
                '0' => 'Disable Self-Registration using QR Code',
                '1' => 'Enable Self-Registration using QR Code',
            ])->displayUsingLabels()->rules('required')->hideFromIndex()->dependsOn(
                ['type'],
                function (Select $field, NovaRequest $request, FormData $formData) {
                    if (request()->user()->hasRole('Administrator')) {
                        $field->readonly(false)->rules('required');
                    }else{
                        $field->readonly(true)->nullable();
                    }
                }
            ),
```

If you can also provide a code that you want to work or algorithm, I can help you adjust it to make it work the way you wanted originally, there are many ways to do things in Nova or override Nova behavior, including making your own custom field or tool.

In observers, you can check:
1) Is the value null => set value to XYZ
2) Has the value been changed by an administrator? Change the value to new value

Or you can simply set the default value in MYSQL for the column in general, lots of options to go around nova
